File: client/unity/Assets/Scripts/Assembly-CSharp/DiamondMessageHandler.cs
public class DiamondMessageHandler : IMiniGameMsgHandler
{
	private static DiamondMessageHandler instance = new DiamondMessageHandler();

	public static void onHandler()
	{
		BoardScr.numPlayer = 2;
		BoardListOnScr.type = BoardListOnScr.STYLE_2PLAYER;
		if (onMainMenu.isOngame)
		{
			RoomListOnScr.setName(2, DiamondScr.gI());
		}
		else
		{
			RoomListOnScr.setName((Canvas.iOpenOngame == 0) ? 2 : 0, DiamondScr.gI());
		}
		CasinoMsgHandler.me.miniGameMessageHandler = instance;
	}

	public void onConnectOK()
	{
	}

	public void onConnectionFail()
	{
	}

	public void onDisconnected()
	{
	}

	public void onMessage(Message msg)
	{
		Out.println("msg: " + msg.command);
		sbyte b = -1;
		sbyte b2 = -1;
		b = msg.reader().readByte();
		b2 = msg.reader().readByte();
		if (!BoardScr.setR_B(b, b2))
		{
			return;
		}
		switch (msg.command)
		{
		case 20:
		{
			sbyte b3 = msg.reader().readByte();
			int whoMoveFirst = msg.reader().readInt();
			sbyte[][] array2 = new sbyte[8][];
			for (int l = 0; l < 8; l++)
			{
				array2[l] = new sbyte[8];
			}
			for (int m = 0; m < 8; m++)
			{
				for (int n = 0; n < 8; n++)
				{
					array2[m][n] = msg.reader().readByte();
				}
			}
			for (int num = 0; num < 2; num++)
			{
				int id = msg.reader().readInt();
				Avatar avatarByID = BoardScr.getAvatarByID(id);
				avatarByID.defence = msg.reader().readShort();
				avatarByID.plusHP = (avatarByID.plusMP = 0);
				avatarByID.hp = (avatarByID.maxHP = msg.reader().readShort());
				avatarByID.mp = msg.reader().readShort();
				avatarByID.maxMP = msg.reader().readShort();
				avatarByID.v *= 2;
				avatarByID.setFeel(4);
			}
			DiamondScr.gI().start(whoMoveFirst, b3, array2);
			break;
		}
		case 64:
		{
			sbyte b6 = msg.reader().readByte();
			sbyte[] array4 = new sbyte[b6];
			AvPosition[] array5 = new AvPosition[b6];
			for (int num7 = 0; num7 < b6; num7++)
			{
				array5[num7] = new AvPosition();
				array4[num7] = msg.reader().readByte();
				array5[num7].anchor = msg.reader().readByte();
				array5[num7].depth = msg.reader().readByte();
			}
			sbyte countCombo = msg.reader().readByte();
			sbyte b7 = msg.reader().readByte();
			MyVector myVector2 = new MyVector();
			for (int num8 = 0; num8 < b7; num8++)
			{
				string o = msg.reader().readUTF();
				myVector2.addElement(o);
			}
			for (int num9 = 0; num9 < 2; num9++)
			{
				int id3 = msg.reader().readInt();
				Avatar avatarByID3 = BoardScr.getAvatarByID(id3);
				avatarByID3.fight = msg.reader().readByte();
				avatarByID3.defence = msg.reader().readShort();
				avatarByID3.plusHP = (short)(msg.reader().readShort() - avatarByID3.hp);
				avatarByID3.plusMP = (short)(msg.reader().readShort() - avatarByID3.mp);
				avatarByID3.isNo = msg.reader().readBoolean();
				if (avatarByID3.isNo)
				{
					DiamondScr.gI().isNo = true;
				}
			}
			DiamondScr.gI().onCreateCell(array4, array5, countCombo, myVector2);
			break;
		}
		case 21:
		{
			int whoMove3 = msg.reader().readInt();
			sbyte b4 = msg.reader().readByte();
			sbyte b5 = msg.reader().readByte();
			DiamondScr.gI().move(whoMove3, b4, b5);
			break;
		}
		case 49:
		{
			int whoMove2 = msg.reader().readInt();
			DiamondScr.gI().onSkip(whoMove2);
			break;
		}
		case 24:
		{
			int whoMove = msg.reader().readInt();
			sbyte[][] array3 = new sbyte[8][];
			for (int num2 = 0; num2 < 8; num2++)
			{
				array3[num2] = new sbyte[8];
			}
			for (int num3 = 0; num3 < 8; num3++)
			{
				for (int num4 = 0; num4 < 8; num4++)
				{
					array3[num3][num4] = msg.reader().readByte();
				}
			}
			DiamondScr.gI().onOutPath(whoMove, array3);
			break;
		}
		case 51:
		{
			MyVector myVector = new MyVector();
			for (int num5 = 0; num5 < 2; num5++)
			{
				int id2 = msg.reader().readInt();
				int num6 = msg.reader().readInt();
				Avatar avatarByID2 = BoardScr.getAvatarByID(id2);
				avatarByID2.v /= 2;
				avatarByID2.action = 0;
				avatarByID2.setMoneyNew(avatarByID2.getMoneyNew() + num6);
				if (num6 != 0)
				{
					Canvas.addFlyText(num6, avatarByID2.x, avatarByID2.y, -1, 30);
					string text = avatarByID2.name + ": ";
					if (num6 > 0)
					{
						DiamondScr.gI().idWin = avatarByID2.IDDB;
						string text2 = text;
						text = text2 + T.win + "   +" + num6 + T.xu;
					}
					else
					{
						string text2 = text;
						text = text2 + T.lose + "  " + num6 + T.xu;
					}
					myVector.addElement("  ");
					myVector.addElement(text);
				}
			}
			DiamondScr.gI().onFinish(myVector);
			break;
		}
		case 71:
		{
			sbyte[][] array = new sbyte[8][];
			for (int i = 0; i < 8; i++)
			{
				array[i] = new sbyte[8];
			}
			for (int j = 0; j < 8; j++)
			{
				for (int k = 0; k < 8; k++)
				{
					array[j][k] = msg.reader().readByte();
				}
			}
			DiamondScr.gI().onData(array);
			break;
		}
		}
	}
}

File: client/java/src/avt/DiamondMessageHandler.java
package avt;

import java.util.Vector;
import main.Canvas;

public final class DiamondMessageHandler extends IService implements IMiniGameMsgHandler {
   private static DiamondMessageHandler instance = new DiamondMessageHandler();

   public static void init() {
      BoardScr.numPlayer = 2;
      BoardListOnScr.type = 0;
      RoomListOnScr.setName(2, DiamondScr.gI());
      CasinoMsgHandler.me.miniGameMessageHandler = instance;
   }

   public final void onMessage(Message var1) {
      try {
         int var2 = var1.reader().readByte();
         int var3 = var1.reader().readByte();
         if (BoardScr.setR_B((byte)var2, (byte)var3)) {
            Vector var4;
            byte var11;
            byte var12;
            int var18;
            int var21;
            int var14;
            String var19;
            switch (var1.command) {
               case 20:
                  var12 = var1.reader().readByte();
                  var3 = var1.reader().readInt();
                  byte[][] var17 = new byte[8][8];

                  for(var18 = 0; var18 < 8; ++var18) {
                     for(var21 = 0; var21 < 8; ++var21) {
                        var17[var18][var21] = var1.reader().readByte();
                     }
                  }

                  for(var18 = 0; var18 < 2; ++var18) {
                     Avatar var20;
                     (var20 = BoardScr.getAvatarByID(var1.reader().readInt())).an = var1.reader().readShort();
                     var20.plusHP = var20.plusMP = 0;
                     var20.hp = var20.maxHP = var1.reader().readShort();
                     var20.mp = var1.reader().readShort();
                     var20.maxMP = var1.reader().readShort();
                     var20.v <<= 1;
                     var20.setFeel(4);
                  }

                  DiamondScr.gI().start(var3, var12, var17);
                  return;
               case 21:
                  var18 = var1.reader().readInt();
                  var21 = var1.reader().readByte();
                  var11 = var1.reader().readByte();
                  DiamondScr.gI().move(var18, var21, var11);
                  return;
               case 24:
                  var2 = var1.reader().readInt();
                  byte[][] var15 = new byte[8][8];

                  for(var14 = 0; var14 < 8; ++var14) {
                     for(var18 = 0; var18 < 8; ++var18) {
                        var15[var14][var18] = var1.reader().readByte();
                     }
                  }

                  DiamondScr.gI().onOutPath(var2, var15);
                  return;
               case 49:
                  var14 = var1.reader().readInt();
                  DiamondScr.gI().onSkip(var14);
                  return;
               case 51:
                  var4 = new Vector();

                  for(var18 = 0; var18 < 2; ++var18) {
                     var2 = var1.reader().readInt();
                     var3 = var1.reader().readInt();
                     Avatar var16;
                     Avatar var10000 = var16 = BoardScr.getAvatarByID(var2);
                     var10000.v /= 2;
                     var16.action = 0;
                     var16.setMoneyNew(var16.getMoneyNew() + var3);
                     if (var3 != 0) {
                        Canvas.addFlyText(var3, var16.x, var16.y, -1, 30);
                        var19 = var16.name + ": ";
                        if (var3 > 0) {
                           DiamondScr.gI().idWin = var16.IDDB;
                           var19 = var19 + T.win + "   +" + var3 + T.xu;
                        } else {
                           var19 = var19 + T.lose + "  " + var3 + T.xu;
                        }

                        var4.addElement("  ");
                        var4.addElement(var19);
                     }
                  }

                  DiamondScr.gI().onFinish(var4);
                  return;
               case 64:
                  byte[] var6 = new byte[var18 = var1.reader().readByte()];
                  AvPosition[] var7 = new AvPosition[var18];

                  for(var2 = 0; var2 < var18; ++var2) {
                     var7[var2] = new AvPosition();
                     var6[var2] = var1.reader().readByte();
                     var7[var2].anchor = var1.reader().readByte();
                     var7[var2].depth = (short)var1.reader().readByte();
                  }

                  var12 = var1.reader().readByte();
                  var11 = var1.reader().readByte();
                  var4 = new Vector();

                  for(var18 = 0; var18 < var11; ++var18) {
                     var19 = var1.reader().readUTF();
                     var4.addElement(var19);
                  }

                  for(var18 = 0; var18 < 2; ++var18) {
                     Avatar var13;
                     (var13 = BoardScr.getAvatarByID(var1.reader().readInt())).fight = var1.reader().readByte();
                     var13.an = var1.reader().readShort();
                     var13.plusHP = (short)(var1.reader().readShort() - var13.hp);
                     var13.plusMP = (short)(var1.reader().readShort() - var13.mp);
                     var13.isNo = var1.reader().readBoolean();
                     if (var13.isNo) {
                        DiamondScr.gI().c = true;
                     }
                  }

                  DiamondScr.gI().paintGame(var6, var7, var12, var4);
                  return;
               case 71:
                  byte[][] var5 = new byte[8][];

                  for(var2 = 0; var2 < 8; ++var2) {
                     var5[var2] = new byte[8];
                  }

                  for(var2 = 0; var2 < 8; ++var2) {
                     for(var3 = 0; var3 < 8; ++var3) {
                        var5[var2][var3] = var1.reader().readByte();
                     }
                  }

                  DiamondScr.gI().onData(var5);
                  return;
            }
         }
      } catch (Exception var15) {
         var15.printStackTrace();
      }

   }
}
